mountpoint-s3-client/src/error_metadata.rs

```diff
@@ -0,0 +1,15 @@
+#[derive(Debug, Clone)]
+pub struct ErrorMetadata {
+    pub http_code: Option<i32>,
+    pub s3_error_code: Option<String>,
+}
+
+pub trait ProvideErrorMetadata {
+    fn meta(&self) -> &ErrorMetadata;
+}
+
+/// Empty error metadata
+pub const EMPTY_ERROR_METADATA: ErrorMetadata = ErrorMetadata {
+    http_code: None,
+    s3_error_code: None,
+};
```

Comment:
just in case, it can not be (which would be a bit simpler):
```
pub const EMPTY: Self = Default::default();
```
seemingly, `default()` is non-const:
```
cannot call non-const fn `<ErrorMetadata as std::default::Default>::default` in constants
```